rainier-core/src/main/scala/com/stripe/rainier/compute/Decimal.scala

```diff
@@ -1,47 +1,129 @@
 package com.stripe.rainier.compute
 
-case class Decimal(value: BigDecimal) {
-  def toDouble = value.toDouble
-  def toInt = value.toInt
+import scala.annotation.tailrec
 
-  def isValidInt = value.isValidInt
-  def isWhole = value.isWhole
+sealed trait Decimal {
+  def ==(other: Decimal) =
```

Comment:
what do you think of return types on the public API?

Reply:
I am in favor of them :) Missed this one, willfix.